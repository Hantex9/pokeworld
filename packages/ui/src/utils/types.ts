/**
 * Enhance the type with the testID that should be used to locate a view in end-to-end tests.
 */
export type TestID = { testID?: string };

export type WithTestID<T> = T & TestID;
/**
 * Ensure that all the keys of type T are required, transforming all optional field of kind T | undefined to T
 */
export type RequiredAll<T> = { [K in keyof T]-?: T[K] };

/**
 * Return a type that prohibits the use of keys that are present only in T but not in U
 */
// eslint-disable-next-line @typescript-eslint/no-unused-vars
type Without<T, U> = { [P in Exclude<keyof T, keyof U>]?: never };

/**
 * Ensure that the types T and U are mutually exclusive
 */
export type XOR<T, U> = T | U extends object ? (Without<T, U> & U) | (Without<U, T> & T) : T | U;

/**
 * Returns a type with the desired type or null
 */
export type Nullable<T> = T | null;

/**
 * Returns a type with the desired type or undefined
 */
export type Optional<T> = T | undefined;

export type TextInputValidationRefProps = { validateInput: () => void };
