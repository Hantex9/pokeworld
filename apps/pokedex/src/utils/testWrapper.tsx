import { ThemeContextProvider } from '@pokeworld/ui';
import { DarkTheme, ThemeProvider } from '@react-navigation/native';
import { render } from '@testing-library/react-native';
import { RelayEnvironmentProvider } from 'react-relay';

const mockEnvironment = {} as any;

export const TestWrapper = ({ children }: { children: React.ReactNode }) => {
  return (
    <RelayEnvironmentProvider environment={mockEnvironment}>
      <ThemeProvider value={DarkTheme}>
        <ThemeContextProvider theme="dark">{children}</ThemeContextProvider>
      </ThemeProvider>
    </RelayEnvironmentProvider>
  );
};

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, { wrapper: TestWrapper, ...options });

export * from '@testing-library/react-native';
export { customRender as render };
