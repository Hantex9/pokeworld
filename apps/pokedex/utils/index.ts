import { POKEMON_STATS_NAME_MAP, POKEMON_TYPES_COLOR_MAP } from '@/constants';
import { PokemonStats, PokemonTypes } from '@/types';

export const getPokemonTypeColor = (pokemonType?: PokemonTypes | undefined | null) => {
  if (!pokemonType) {
    return POKEMON_TYPES_COLOR_MAP['unknown'];
  }
  return POKEMON_TYPES_COLOR_MAP[pokemonType];
};

export const getPokemonStatName = (stat?: PokemonStats | undefined | null) => {
  if (!stat) {
    return 'Unknown';
  }
  return POKEMON_STATS_NAME_MAP[stat];
};

export const toDecimal = (num: number): number => {
  if (typeof num !== 'number' || isNaN(num)) return NaN;
  const numStr = num.toString();
  if (numStr.length === 1) {
    return num / 10;
  }
  const intPart = numStr.slice(0, -1);
  const decimalPart = numStr.slice(-1);
  return parseFloat(`${intPart}.${decimalPart}`);
};

export const padToThreeDigits = (num: number): string => {
  return num.toString().padStart(3, '0');
};
