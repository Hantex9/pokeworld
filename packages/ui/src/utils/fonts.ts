/**
 * Utility functions to manage font properties to style mapping for both iOS and Android
 * Fonts are managed differently on Android and iOS.
 */

import { Platform, TextStyle } from 'react-native';

/**
 * Choose the font name based on the platform
 */
const fonts = {
  TitilliumSansPro: Platform.select({
    android: 'TitilliumSansPro',
    web: 'TitilliumSansPro',
    ios: 'Titillium Sans Pro',
    default: 'TitilliumSansPro',
  }),
  Titillio: Platform.select({
    android: 'Titillio',
    web: 'Titillio',
    ios: 'Titillio',
    default: 'Titillio',
  }),
  DMMono: Platform.select({
    android: 'DMMono',
    web: 'DMMono',
    ios: 'DM Mono',
    default: 'DMMono',
  }),
} as const;

export type FontFamily = keyof typeof fonts;

/*
 * Font Sizes
 */
const fontSizes = [12, 14, 16, 20, 22, 26, 28, 32] as const;
const fontSizesLegacy = [17, 28, 31, 35] as const;
const allFontSizes = [...new Set([...fontSizes, ...fontSizesLegacy])];
export type FontSize = (typeof allFontSizes)[number];

/*
 * Font Weights
 */

const weights = ['Thin', 'Light', 'Regular', 'Medium', 'Semibold', 'Bold', 'Black'] as const;
export type FontWeight = (typeof weights)[number];

const weightValues = ['200', '300', '400', '500', '600', '700', '900'] as const;
export type FontWeightNumeric = (typeof weightValues)[number];

/**
 * Mapping between the nominal description of the weight (also the postfix used on Android) and the numeric value
 * used on iOS
 */
export const fontWeights: Record<FontWeight, FontWeightNumeric> = {
  Thin: '200',
  Light: '300',
  Regular: '400',
  Medium: '500',
  Semibold: '600',
  Bold: '700',
  Black: '900',
};

type FontStyleObject = {
  fontSize: FontSize | number;
  /* We also accept `string` because Android needs a composed 
  fontFamily name, like `TitilliumSansPro-Regular` */
  fontFamily: string | FontFamily;
  fontWeight?: FontWeightNumeric;
  lineHeight?: TextStyle['lineHeight'];
  fontStyle?: TextStyle['fontStyle'];
  boldEnabled?: boolean;
};

/* Function that, given a certain weight, returns the next
available `FontWeight` value. 
For example, if I set it to `Regular`, the function 
should return `Medium`, and so on. If I set it to the last `FontWeight`
value, the function will return the same value.
*/
const getBolderFontWeight = (weight: FontWeight): FontWeight => {
  const currentWeight = weights.indexOf(weight);
  return currentWeight === weights.length - 1 ? weight : weights[currentWeight + 1];
};

/**
 * Get the correct `fontFamily` name on both Android and iOS.
 * @param font
 * @param weight
 * @param isItalic
 */
export const makeFontFamilyName = (
  font: FontFamily,
  weight: FontWeight = defaultWeight,
  fontStyle: TextStyle['fontStyle'] = 'normal'
): string =>
  Platform.select({
    web: fonts[font],
    android: `${fonts[font]}-${weight || 'Regular'}${fontStyle === 'italic' ? 'Italic' : ''}`,
    ios: fonts[font],
    default: fonts[font],
  });

/**
 * Default `Text` typography style
 */
const defaultFont: FontFamily = 'TitilliumSansPro';
const defaultWeight: FontWeight = 'Regular';
const defaultFontSize: FontSize = 16;
export const MaxFontSizeMultiplier = 1.5;

/**
 * Return a {@link FontStyleObject} with the fields filled based on the platform (iOS or Android).
 * @param size
 * @param font
 * @param weight
 * @param fontStyle
 */

export const makeFontStyleObject = (
  size: number = defaultFontSize,
  font: FontFamily = defaultFont,
  lineHeight: TextStyle['lineHeight'],
  weight: FontWeight = defaultWeight,
  fontStyle: TextStyle['fontStyle'] = 'normal',
  boldEnabled: boolean = false
): FontStyleObject => {
  const dynamicWeight = boldEnabled ? getBolderFontWeight(weight) : weight;

  return Platform.select({
    web: {
      fontSize: size,
      fontFamily: makeFontFamilyName(font, dynamicWeight, fontStyle),
      fontWeight: fontWeights[dynamicWeight],
      lineHeight,
      fontStyle,
    },
    android: {
      fontSize: size,
      fontFamily: makeFontFamilyName(font, dynamicWeight, fontStyle),
      lineHeight,
      includeFontPadding: false,
    },
    ios: {
      fontSize: size,
      fontFamily: makeFontFamilyName(font, dynamicWeight, fontStyle),
      fontWeight: fontWeights[dynamicWeight],
      lineHeight,
      fontStyle,
    },
    default: {
      fontSize: size,
      fontFamily: makeFontFamilyName(font, dynamicWeight, fontStyle),
    },
  });
};
