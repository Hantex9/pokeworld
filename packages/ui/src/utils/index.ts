import { ColorValue } from 'react-native';

export const hexToRgba = (hexCode: ColorValue, opacity: number = 1) => {
  const hex = hexCode.toString().replace('#', '');

  const r = parseInt(hex.substring(0, 2), 16);
  const g = parseInt(hex.substring(2, 4), 16);
  const b = parseInt(hex.substring(4, 6), 16);

  return `rgba(${r},${g},${b},${opacity})`;
};
