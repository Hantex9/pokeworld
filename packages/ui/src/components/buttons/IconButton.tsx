import { GestureResponderEvent, Pressable } from 'react-native';
import Animated, {
  interpolateColor,
  useAnimatedProps,
  useReducedMotion,
} from 'react-native-reanimated';

import { useTheme } from '../../context';
import { Colors, IconButtonStyles, ThemeLight } from '../../core';
import { useScaleAnimation } from '../../hooks';
import { hexToRgba } from '../../utils';
import { WithTestID } from '../../utils/types';
import { AnimatedIcon, AnimatedIconWithColorTransition, IconSizeScale, Icons } from '../icons';

export type IconButton = WithTestID<{
  color?: 'primary' | 'neutral' | 'contrast';
  icon: Icons;
  iconSize?: IconSizeScale;
  disabled?: boolean;
  accessibilityLabel: string;
  accessibilityHint?: string;
  onPress: (event: GestureResponderEvent) => void;
  persistentColorMode?: boolean;
}>;

type ColorStates = {
  icon: {
    default: string;
    pressed: string;
    disabled: string;
  };
};

export const IconButton = ({
  color = 'primary',
  persistentColorMode = false,
  icon,
  iconSize = 24,
  disabled = false,
  onPress,
  accessibilityLabel,
  accessibilityHint,
  testID,
}: IconButton) => {
  const theme = useTheme();
  const { progress, onPressIn, onPressOut, scaleAnimatedStyle } = useScaleAnimation('exaggerated');
  const reducedMotion = useReducedMotion();

  const mapColorStates: Record<NonNullable<IconButton['color']>, ColorStates> = {
    // Primary button
    primary: {
      icon: {
        default: Colors[theme['interactiveElem-default']],
        pressed: Colors[theme['interactiveElem-pressed']],
        disabled: Colors[theme['interactiveElem-disabled']],
      },
    },
    // Neutral button
    neutral: {
      icon: {
        default: persistentColorMode
          ? Colors[ThemeLight['neutralButton-default']]
          : Colors[theme['neutralButton-default']],
        pressed: persistentColorMode
          ? Colors[ThemeLight['neutralButton-pressed']]
          : Colors[theme['neutralButton-pressed']],
        disabled: persistentColorMode
          ? Colors[ThemeLight['neutralButton-disabled']]
          : Colors[theme['neutralButton-disabled']],
      },
    },
    // Contrast button
    contrast: {
      icon: {
        default: Colors.white,
        pressed: hexToRgba(Colors.white, 0.85),
        disabled: hexToRgba(Colors.white, 0.25),
      },
    },
  };

  // Animate the <Icon> color prop
  const animatedColor = useAnimatedProps(() => {
    const iconColor = interpolateColor(
      progress.value,
      [0, 1],
      [mapColorStates[color].icon.default, mapColorStates[color].icon.pressed]
    );
    return { color: iconColor };
  });

  return (
    <Pressable
      disabled={disabled}
      // Events
      onPress={onPress}
      onPressIn={onPressIn}
      onPressOut={onPressOut}
      // Accessibility
      accessible={true}
      accessibilityRole={'button'}
      accessibilityLabel={accessibilityLabel}
      accessibilityHint={accessibilityHint}
      accessibilityState={{ disabled }}
      // Usability
      // Add a touchable area around the button
      hitSlop={8}
      // Test
      testID={testID}
    >
      <Animated.View
        style={[
          IconButtonStyles.button,
          IconButtonStyles.buttonSizeSmall,
          !disabled && !reducedMotion && scaleAnimatedStyle,
        ]}
      >
        {!disabled ? (
          <AnimatedIconWithColorTransition
            allowFontScaling
            name={icon}
            size={iconSize}
            animatedProps={animatedColor}
            color={mapColorStates[color]?.icon?.default}
          />
        ) : (
          <AnimatedIcon
            allowFontScaling
            name={icon}
            size={iconSize}
            color={mapColorStates[color]?.icon?.disabled}
          />
        )}
      </Animated.View>
    </Pressable>
  );
};

export default IconButton;
