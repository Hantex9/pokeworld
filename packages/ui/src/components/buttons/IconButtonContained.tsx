import { GestureResponderEvent, Pressable } from 'react-native';
import Animated, {
  interpolateColor,
  useAnimatedProps,
  useAnimatedStyle,
  useReducedMotion,
} from 'react-native-reanimated';

import { AnimatedIcon, AnimatedIconWithColorTransition, Icons } from '../../components/icons';
import { Colors, IconButtonStyles } from '../../core';
import { useScaleAnimation } from '../../hooks';
import { hexToRgba } from '../../utils';
import { WithTestID } from '../../utils/types';

export type IconButtonContained = WithTestID<{
  icon: Icons;
  color?: 'primary' | 'neutral' | 'contrast';
  disabled?: boolean;
  accessibilityLabel: string;
  accessibilityHint?: string;
  onPress: (event: GestureResponderEvent) => void;
}>;

type ColorStates = {
  background: {
    default: string;
    pressed: string;
    disabled: string;
  };
  icon: {
    default: string;
    pressed: string;
    disabled: string;
  };
};

const mapColorStates: Record<NonNullable<IconButtonContained['color']>, ColorStates> = {
  // Primary button
  primary: {
    background: {
      default: hexToRgba(Colors['blue-500'], 0),
      pressed: hexToRgba(Colors['blue-500'], 0.15),
      disabled: 'transparent',
    },
    icon: {
      default: Colors['blue-500'],
      pressed: Colors['blue-600'],
      disabled: hexToRgba(Colors['blue-500'], 0.25),
    },
  },
  // Neutral button
  neutral: {
    background: {
      default: Colors.white,
      pressed: Colors['grey-50'],
      disabled: 'transparent',
    },
    icon: {
      default: Colors['grey-700'],
      pressed: Colors.black,
      disabled: Colors['grey-450'],
    },
  },
  // Contrast button
  contrast: {
    background: {
      default: hexToRgba(Colors.white, 0),
      pressed: hexToRgba(Colors.white, 0.2),
      disabled: 'transparent',
    },
    icon: {
      default: Colors.white,
      pressed: Colors.white,
      disabled: hexToRgba(Colors.white, 0.25),
    },
  },
};

export const IconButtonContained = ({
  icon,
  color = 'primary',
  disabled = false,
  onPress,
  accessibilityLabel,
  accessibilityHint,
  testID,
}: IconButtonContained) => {
  const { progress, onPressIn, onPressOut, scaleAnimatedStyle } = useScaleAnimation('exaggerated');
  const reducedMotion = useReducedMotion();

  const backgroundColorAnimationStyle = useAnimatedStyle(() => ({
    backgroundColor: interpolateColor(
      progress.value,
      [0, 1],
      [mapColorStates[color].background.default, mapColorStates[color].background.pressed]
    ),
  }));

  // Animate the <Icon> color prop
  const iconColorAnimationStyle = useAnimatedProps(() => ({
    color: interpolateColor(
      progress.value,
      [0, 1],
      [mapColorStates[color].icon.default, mapColorStates[color].icon.pressed]
    ),
  }));

  return (
    <Pressable
      accessibilityLabel={accessibilityLabel}
      accessibilityHint={accessibilityHint}
      accessibilityRole={'button'}
      accessibilityState={{ disabled }}
      testID={testID}
      onPress={onPress}
      onPressIn={onPressIn}
      onPressOut={onPressOut}
      accessible={true}
      disabled={disabled}
      style={{ alignSelf: 'flex-start' }}
    >
      <Animated.View
        style={[
          IconButtonStyles.button,
          IconButtonStyles.buttonSizeDefault,
          !disabled && !reducedMotion && scaleAnimatedStyle,
          !disabled && backgroundColorAnimationStyle,
        ]}
      >
        {!disabled ? (
          <AnimatedIconWithColorTransition
            name={icon}
            animatedProps={iconColorAnimationStyle}
            color={mapColorStates[color]?.icon?.default}
          />
        ) : (
          <AnimatedIcon name={icon} color={mapColorStates[color]?.icon?.disabled} />
        )}
      </Animated.View>
    </Pressable>
  );
};

export default IconButtonContained;
