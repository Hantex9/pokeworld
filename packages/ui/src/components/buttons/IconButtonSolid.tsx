import { GestureResponderEvent, Pressable } from 'react-native';
import Animated, {
  interpolateColor,
  useAnimatedStyle,
  useReducedMotion,
} from 'react-native-reanimated';

import { useTheme } from '../../context';
import { IconButtonStyles } from '../../core';
import { Colors } from '../../core/Colors';
import { useScaleAnimation } from '../../hooks';
import { hexToRgba } from '../../utils';
import { WithTestID } from '../../utils/types';
import { AnimatedIcon, Icons } from '../icons';

export type IconButtonSolid = WithTestID<{
  icon: Icons;
  color?: 'primary' | 'contrast';
  disabled?: boolean;
  accessibilityLabel: string;
  accessibilityHint?: string;
  onPress: (event: GestureResponderEvent) => void;
}>;

type ColorStates = {
  background: {
    default: string;
    pressed: string;
    disabled: string;
  };
  icon: {
    default: string;
    disabled: string;
  };
};

export const IconButtonSolid = ({
  icon,
  color = 'primary',
  disabled = false,
  onPress,
  accessibilityLabel,
  accessibilityHint,
  testID,
}: IconButtonSolid) => {
  const theme = useTheme();
  const { progress, onPressIn, onPressOut, scaleAnimatedStyle } = useScaleAnimation('exaggerated');
  const reducedMotion = useReducedMotion();

  const mapColorStates: Record<NonNullable<IconButtonSolid['color']>, ColorStates> = {
    // Primary button
    primary: {
      background: {
        default: Colors[theme['interactiveElem-default']],
        pressed: Colors[theme['interactiveElem-pressed']],
        disabled: Colors[theme['interactiveElem-disabled']],
      },
      icon: {
        default: Colors[theme['buttonText-default']],
        disabled: Colors[theme['buttonText-disabled']],
      },
    },
    contrast: {
      background: {
        default: Colors.white,
        pressed: Colors['blue-50'],
        disabled: hexToRgba(Colors.white, 0.25),
      },
      icon: {
        default: Colors['blue-500'],
        disabled: Colors['blue-500'],
      },
    },
  };

  const backgroundColorAnimationStyle = useAnimatedStyle(() => ({
    backgroundColor: interpolateColor(
      progress.value,
      [0, 1],
      [mapColorStates[color].background.default, mapColorStates[color].background.pressed]
    ),
  }));

  return (
    <Pressable
      accessibilityLabel={accessibilityLabel}
      accessibilityHint={accessibilityHint}
      accessibilityRole={'button'}
      accessibilityState={{ disabled }}
      testID={testID}
      onPress={onPress}
      onPressIn={onPressIn}
      onPressOut={onPressOut}
      accessible={true}
      disabled={disabled}
      style={{ alignSelf: 'flex-start' }}
    >
      <Animated.View
        style={[
          IconButtonStyles.button,
          IconButtonStyles.buttonSizeLarge,
          !disabled && !reducedMotion && scaleAnimatedStyle,
          !disabled && backgroundColorAnimationStyle,
          disabled
            ? { backgroundColor: mapColorStates[color]?.background?.disabled }
            : { backgroundColor: mapColorStates[color]?.background?.default },
        ]}
      >
        <AnimatedIcon
          allowFontScaling
          name={icon}
          color={
            !disabled ? mapColorStates[color]?.icon?.default : mapColorStates[color]?.icon?.disabled
          }
        />
      </Animated.View>
    </Pressable>
  );
};

export default IconButtonSolid;
