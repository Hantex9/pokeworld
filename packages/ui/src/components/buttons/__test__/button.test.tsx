import React from "react";
import { Alert } from "react-native";
import * as TestRenderer from "react-test-renderer";
import { TestRendererWithExperimentalEnabledContextProvider } from "../../../utils/testing";
import ButtonLink from "../ButtonLink";
import ButtonOutline from "../ButtonOutline";
import ButtonSolid from "../ButtonSolid";
import IconButton from "../IconButton";
import IconButtonContained from "../IconButtonContained";
import IconButtonSolid from "../IconButtonSolid";

const onButtonPress = () => {
  Alert.alert("Alert", "Action triggered");
};

describe("Test Buttons Components", () => {
  it("ButtonSolid Snapshot", () => {
    const buttonSolid = TestRenderer.create(
      <ButtonSolid
        label={"label"}
        accessibilityLabel={"accessibilityLabel"}
        onPress={onButtonPress}
      ></ButtonSolid>
    ).toJSON();
    expect(buttonSolid).toMatchSnapshot();
  });

  it("ButtonLink Snapshot", () => {
    const buttonLink = TestRenderer.create(
      <ButtonLink label={"label"} onPress={onButtonPress}></ButtonLink>
    ).toJSON();
    expect(buttonLink).toMatchSnapshot();
  });

  it("ButtonOutline Snapshot", () => {
    const buttonOutline = TestRenderer.create(
      <ButtonOutline
        label={"label"}
        accessibilityLabel={"accessibilityLabel"}
        onPress={onButtonPress}
      ></ButtonOutline>
    ).toJSON();
    expect(buttonOutline).toMatchSnapshot();
  });
});

describe("Test Buttons Components - Experimental Enabled", () => {
  it("ButtonSolid Snapshot", () => {
    const buttonSolid = TestRendererWithExperimentalEnabledContextProvider(
      <ButtonSolid
        label={"label"}
        accessibilityLabel={"accessibilityLabel"}
        onPress={onButtonPress}
      ></ButtonSolid>
    ).toJSON();
    expect(buttonSolid).toMatchSnapshot();
  });

  it("ButtonLink Snapshot", () => {
    const buttonLink = TestRendererWithExperimentalEnabledContextProvider(
      <ButtonLink label={"label"} onPress={onButtonPress}></ButtonLink>
    ).toJSON();
    expect(buttonLink).toMatchSnapshot();
  });

  it("ButtonOutline Snapshot", () => {
    const buttonOutline = TestRendererWithExperimentalEnabledContextProvider(
      <ButtonOutline
        label={"label"}
        accessibilityLabel={"accessibilityLabel"}
        onPress={onButtonPress}
      ></ButtonOutline>
    ).toJSON();
    expect(buttonOutline).toMatchSnapshot();
  });

  it("IconButtonSolid Snapshot", () => {
    const iconButtonSolid = TestRendererWithExperimentalEnabledContextProvider(
      <IconButtonSolid
        onPress={onButtonPress}
        icon={"spid"}
        accessibilityLabel={"accessibilityLabel"}
      ></IconButtonSolid>
    ).toJSON();
    expect(iconButtonSolid).toMatchSnapshot();
  });

  it("IconButton Snapshot", () => {
    const iconButton = TestRendererWithExperimentalEnabledContextProvider(
      <IconButton
        onPress={onButtonPress}
        icon={"spid"}
        accessibilityLabel={"accessibilityLabel"}
      ></IconButton>
    ).toJSON();
    expect(iconButton).toMatchSnapshot();
  });

  it("IconButtonContained Snapshot", () => {
    const iconButtonContained =
      TestRendererWithExperimentalEnabledContextProvider(
        <IconButtonContained
          onPress={onButtonPress}
          icon={"spid"}
          accessibilityLabel={"accessibilityLabel"}
        ></IconButtonContained>
      ).toJSON();
    expect(iconButtonContained).toMatchSnapshot();
  });
});
