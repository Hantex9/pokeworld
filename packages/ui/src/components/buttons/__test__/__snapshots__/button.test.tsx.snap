// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Buttons Components - Experimental Enabled ButtonLink Snapshot 1`] = `null`;

exports[`Test Buttons Components - Experimental Enabled ButtonOutline Snapshot 1`] = `null`;

exports[`Test Buttons Components - Experimental Enabled ButtonSolid Snapshot 1`] = `null`;

exports[`Test Buttons Components - Experimental Enabled IconButton Snapshot 1`] = `null`;

exports[`Test Buttons Components - Experimental Enabled IconButtonContained Snapshot 1`] = `null`;

exports[`Test Buttons Components - Experimental Enabled IconButtonSolid Snapshot 1`] = `null`;

exports[`Test Buttons Components ButtonLink Snapshot 1`] = `null`;

exports[`Test Buttons Components ButtonOutline Snapshot 1`] = `null`;

exports[`Test Buttons Components ButtonSolid Snapshot 1`] = `null`;
